File: app/src/main/java/com/student/Utils/CustomDialogFragment.java
package com.student.Utils;

import android.content.Context;
import android.content.res.Resources;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import com.bumptech.glide.load.engine.Resource;
import com.student.models.ProfileData;
import com.student.rentals.R;
import com.student.rentals.ui.fragments.view_fragment.ViewItemViewModel;

import timber.log.Timber;

/**
 * Copyright (c) $today.year.
 * This file was created by eli on 1/28/20
 * You are not allowed to copy it or use it in another project
 * Without permission from creator
 **/
public class CustomDialogFragment extends DialogFragment {
    private Resources resources = getActivity().getResources();
    private ViewItemViewModel viewModel;

    @Override
    public void onAttach(@NonNull final Context context) {
        super.onAttach(context);
        Timber.d(resources.getString(R.string.on_attach));
    }
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        if (savedInstanceState != null) {

        }


        return inflater.inflate(R.layout.item_property_popup_profile, container, false);
    }


    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this).get(ViewItemViewModel.class);
    }



    @Override
    public void onStart() {
        super.onStart();
        Timber.d(resources.getString(R.string.on_start));
    }

    @Override
    public void onStop() {
        super.onStop();
        Timber.d(resources.getString(R.string.on_stop));
    }


    @Override
    public void onResume() {
        super.onResume();
        Timber.d(resources.getString(R.string.on_resume));
        viewModel.getProfileData(1).observe(getViewLifecycleOwner(), new Observer<ProfileData>() {
            @Override
            public void onChanged(ProfileData profileData) {
                Timber.d(profileData.toString());
            }
        });
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onViewStateRestored(@Nullable final Bundle savedInstanceState) {
        super.onViewStateRestored(savedInstanceState);
    }
}

File: app/src/main/java/com/student/Utils/CustomDialogFragment.kt
package com.student.Utils

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.Toast
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentActivity
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.transition.DrawableCrossFadeFactory
import com.student.models.DUserData
import com.student.rentals.R
import com.student.rentals.databinding.ItemPropertyPopupProfileBinding
import kotlinx.android.synthetic.main.item_property_popup_profile.*
import timber.log.Timber

/**
 * Copyright (c) $today.year.
 * This file was created by eli on 1/28/20
 * You are not allowed to copy it or use it in another project
 * Without permission from creator
 */
class CustomDialogFragment : DialogFragment() {
    private var content:String? = null
    private var u:DUserData? = null
    private lateinit var binding: ItemPropertyPopupProfileBinding
    override fun getTheme(): Int {
        return R.style.AppTheme_NoActionBar_FullScreenDialog
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        content = arguments?.getString("content")
        val style = DialogFragment.STYLE_NO_FRAME
        val theme = R.style.DialogTheme
        setStyle(style,theme)
        retainInstance = true
    }
    override fun onResume() {
        super.onResume()

        val params:ViewGroup.LayoutParams = dialog!!.window!!.attributes
        params.width = LinearLayout.LayoutParams.WRAP_CONTENT
        params.height = LinearLayout.LayoutParams.WRAP_CONTENT
        dialog?.window?.attributes = params as android.view.WindowManager.LayoutParams

        popup_close.setOnClickListener {
            Toast.makeText(activity, "Action cancelled", Toast.LENGTH_SHORT).show()
            dismiss()
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        Timber.d(resources.getString(R.string.on_attach))
    }
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = ItemPropertyPopupProfileBinding.inflate(inflater, container, false)
        updateUI(u)

        return binding.root
    }

   companion object{
       fun newInstance(content: String, bundle:Bundle): CustomDialogFragment{
           val f = CustomDialogFragment()
           val args = bundle
           var data = bundle.getParcelable<DUserData>(Constants.PARCEL_KEY)

           args.putString("content", content)
           f.arguments = args
           f.u = data
           return f
       }
   }
   open fun updateUI( u:DUserData?) {

        binding.userData = u
       Timber.d(u.toString())
        val factory = DrawableCrossFadeFactory.Builder().setCrossFadeEnabled(true).build()
       GlideApp
            .with(dialog?.context!!)
            .load(Constants.IMAGE_BASE_URL + u?.thumbNail)
            .transition(DrawableTransitionOptions.withCrossFade(factory))
            .diskCacheStrategy(DiskCacheStrategy.ALL)
            .apply(RequestOptions.circleCropTransform())
            .placeholder(R.drawable.person)
            .error((R.drawable.person))
            .into(binding.root.findViewById(R.id.popup_profile_image))


    }
}